rpcs3/Emu/RSX/GL/GLTexture.cpp

```diff
@@ -319,6 +319,14 @@ namespace rsx
 				std::vector<gsl::byte> staging_buffer)
 			{
 				int mip_level = 0;
+				if (is_compressed_format(format))
+				{
+					//Compressed formats have a 4-byte alignment
+					//TODO: Verify that samplers are not affected by the padding
+					width = align(width, 4);
+					height = align(height, 4);
```

Comment:
- I didn't touch vulkan at all with this PR; just added in support for the arrayelement method.
- Also, align(2, 4) results in a height of 4 pixels while layout.height_in_block should return 1, multiplying by 4 as you initially did gives the same value; the problem is probably caused by something else I did and not alignment, but you may confirm.
- BTW, are the indices passed correct? I am not able to test endianness of the packed value myself, so I need testers to confitm correct behaviour.